sql/catalyst/src/main/scala/org/apache/spark/sql/QueryExecutionErrors.scala

```diff
@@ -51,18 +58,209 @@ object QueryExecutionErrors {
+  }
+
+  def orderedOperationUnsupportedByDataTypeError(dataType: DataType): Throwable = {
+    new IllegalArgumentException(s"Type $dataType does not support ordered operations")
+  }
+
+  def indexOfMatchGroupLessThanZeroNotAllowedError(): Throwable = {
+    new IllegalArgumentException("The specified group index cannot be less than zero")
+  }
+
+  def groupIndexExceedMatchGroupCountNotAllowedError(
+      groupCount: Int, groupIndex: Int): Throwable = {
+    new IllegalArgumentException(
+      s"Regex group count is $groupCount, but the specified group index is $groupIndex")
+  }
+
+  def invalidUrlError(url: UTF8String, e: URISyntaxException): Throwable = {
+    new IllegalArgumentException(s"Find an invaild url string ${url.toString}", e)
+  }
+
+  def dataTypeUnsupportedError(): Throwable = {
+    new UnsupportedOperationException("dataType")
+  }
+
+  def mergeUnsupportedByWindowFunctionError(): Throwable = {
+    new UnsupportedOperationException("Window Functions do not support merging.")
+  }
+
+  def dataTypeUnexpectedError(dataType: DataType): Throwable = {
+    new UnsupportedOperationException(s"Unexpected data type ${dataType.catalogString}")
+  }
+
+  def negativeValueUnexpectedError(frequencyExpression : Expression): Throwable = {
+    new SparkException(s"Negative values found in ${frequencyExpression.sql}")
+  }
+
+  def addNewFunctionMismatchedWithFunctionError(funcName: String): Throwable = {
+    new IllegalArgumentException(s"$funcName is not matched at addNewFunction")
+  }
+
+  def cannotGenerateCodeForUnComparableTypeError(
+      codeType: String, dataType: DataType): Throwable = {
+    new IllegalArgumentException(
+      s"cannot generate $codeType code for un-comparable type: ${dataType.catalogString}")
+  }
+
+  def cannotGenerateCodeForUnsupportedTypeError(dataType: DataType): Throwable = {
+    new IllegalArgumentException(s"cannot generate code for unsupported type: $dataType")
+  }
+
+  def cannotInterpolateClassIntoCodeBlockError(arg: Any): Throwable = {
+    new IllegalArgumentException(
+      s"Can not interpolate ${arg.getClass.getName} into code block.")
+  }
+
+  def customCollectionClsCannotResolvedError(): Throwable = {
+    new UnsupportedOperationException("not resolved")
+  }
+
+  def classUnsupportedByMapObjectError(cls: Class[_]): Throwable = {
+    new RuntimeException(s"class `${cls.getName}` is not supported by `MapObjects` as " +
+      "resulting collection.")
+  }
+
+  def nullAsMapKeyNotAllowedError(): Throwable = {
+    new RuntimeException("Cannot use null as map key!")
+  }
+
+  def methodNotDeclaredError(name: String): Throwable = {
+    new NoSuchMethodException(s"""A method named "$name" is not declared """ +
+      "in any enclosing class nor any supertype")
+  }
+
+  def inputExternalRowCannotBeNullError(): Throwable = {
+    new RuntimeException("The input external row cannot be null.")
+  }
+
+  def fieldOfInputRowCannotBeNullMsg(index: Int, fieldName: String): String = {
```

Comment:
We can remove this and directly use the error message inside the function below.

Reply:
https://github.com/apache/spark/blob/7a607f3a7d0234036b348961db8d597a01a5b024/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala#L1694 references `fieldOfInputRowCannotBeNullMsg`